libraries/plugins/elasticsearch/elasticsearch_plugin.cpp

```diff
@@ -59,6 +58,8 @@ class elasticsearch_plugin_impl
       std::string _elasticsearch_index_prefix = "bitshares-";
       bool _elasticsearch_operation_object = false;
       uint32_t _elasticsearch_start_es_after_block = 0;
+      bool _elasticsearch_operation_string = true;
+      std::string _elasticsearch_mode = "only_save";
```

Comment:
For more efficient comparison, make this an enum type.

Reply:
It is a bit tricky to do this because the `boost::program_options` will not accept the enum but basically just a string, number or boolean. 
I picked the number option to do a static cast to the enum by checking the upper boundary to avoid inserting invalid options.

Implemented at boost::program_options

Reply:
changed the exception introduced here to `graphene::chain::plugin_exception` in the context of https://github.com/bitshares/bitshares-core/pull/1725/commits/5838a38fbc93b2673e7c96640abe15a5caa4533e